### ISSUE TITLE
Horrible migration documentation for anyone getting started with using eslint-stylistic

### ISSUE THREAD

**Issue by kshetline** [eslint-stylistic/eslint-stylistic#830]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

I'd propose more specific changes, but since I'm a frustrated and confused user trying to get started, I don't know enough answers yet to provide specific solutions.

For instance, right from the get-go, you need to say that the `eslint` package is still needed. At the start of the migration page you seem to suggest the opposite when you say:

"ESLint Stylistic is migrated from 3 different sources packages"

...and then list `eslint` as one of those packages, sure making it sound like `eslint` would be _replaced by_ `@stylistic/eslint-plugin`. Only later where you mention needing "ESLint >=v8.40.0" in the FAQ does it become apparent that `@stylistic/eslint-plugin` is meant to be uses _along with_ `eslint` rather than replacing it.

In the Migration Guide, you show a sample "Flat config" file, not mentioning what that flat config file should be named. From the Rules section, only going by a comment in another sample config, I can see that the config file should probably be named `eslint.config.js`... but why make it this hard with these details scattered all around, sure to have people banging there heads against a wall before eventually digging down an finding what they need elsewhere?

Listen, I appreciate the effort that goes into the unpaid work of an open-source project, and I don't want to seem ungrateful. But just a few simple additions to the documentation by people who already know this new plugin way better than a beginner could make the migration process a whole lot less painful.

I intend to press on migrating my first project to `@stylistic/eslint-plugin`, but I'm pretty sure a lot more unnecessary frustration is coming my way before I get this working.

### Reproduction

Documentation comment - reproduction n/a

### Contributes

- [x] I am willing to submit a PR to fix this issue
- [x] I am willing to submit a PR with failing tests

**Comment by antfu** [comment]
Thank you! It would be even greater if you could help by sending a PR to address and help improve those problems. We are certainly biased as we are the maintainers and we know it too well to know that newcomers would have trouble with, we are counting on contributors to help us improvint the docs

**Comment by kshetline** [comment]
I'll gladly try my hand at it, but it will just be from the perspective of a beginner, and can only cover what I had to work through to get my own particular TypeScript project working.

Whatever I write will definitely need maintainer input since I don't really know if I've found _good_ solutions or simply _expedient_ solutions.

**Comment by reteps** [comment]
I just added a table to "eslint-react" that essentially shows the rules that overlap (https://github.com/Rel1cx/eslint-react/pull/1152), as well as points rules to this project. I hope that this project does the same (essentially pulling out the "legacy" config into a table you can see).

**Comment by shaineoneal** [comment]
I'm not sure if this is just a Jetbrains issue, but I had a very similar issue with this.

This is the only configuration I could get to work:

```js
import stylistic from '@stylistic/eslint-plugin'
import typescriptParser from '@typescript-eslint/parser'

export default [
    stylistic.configs.customize({
        semi: true,
        indent: 4,
        jsx: true,
    }),
    {
        files: ['**/*.ts', '**/*.tsx'],
        plugins: {
            '@stylistic': stylistic
        },
        languageOptions: {
            parser: typescriptParser
        }
    }
];
```

I have no idea if this is applicable to anyone else, but using the `rules: {...}` did not work for me at all.
